### PR TITLE
Allow building on macOS

### PR DESCRIPTION
**Pull request by stmcginnis**
Journald is not available on mac. To allow building the rest of the
project while working on a mac, use the same flag as the Windows build
to skip inclusion of journald.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -53,6 +53,10 @@ IMAGE:=$(REGISTRY)/node-problem-detector:$(TAG)
 # support needs libsystemd-dev or libsystemd-journal-dev.
 ENABLE_JOURNALD?=1
 
+ifeq ($(shell uname -s 2>/dev/null), Darwin)
```
Comment (Random-Liu) [inline Makefile:56]: Can we use `go env` instead and also add windows here?

Comment (stmcginnis) [reply Makefile:56]: Great idea, I'll update shortly.

**Comment by stmcginnis** [comment]
/retest

Looks like this may be the issue: https://github.com/kubernetes/node-problem-detector/issues/403

**Comment by gengwg** [comment]
Is there instructions how to build the image on macos? the README is for debian.

**Pull request merged**